### PR TITLE
#173 ensure avatar uploads are resized/filepond implementation

### PR DESCRIPTION
**Pull request by reygcalantaol**

### CODE REVIEW

File: assets/js/wp-user-manager.js [hunk 1]
```diff
@@ -165,4 +167,116 @@ jQuery( function( $ ) {
...
+					metadata: {
+						poster: defaultCover.length > 0 ? defaultCover.val() : null
+					}
+				}
+			} ];
+		}
+
+		const filePondCover = document.querySelector('.fieldset-user_cover .filepond--root');
+		filePondCover.addEventListener('FilePond:removefile', e => {
+			$('.fieldset-user_cover .wpum-uploaded-files').html('');
```
Comment (polevaultweb) [inline assets/js/wp-user-manager.js:270]: Can we add support for any image file field (https://github.com/WPUserManager/wpum-custom-fields/issues/42)

**Review by polevaultweb — CHANGES_REQUESTED** [review]

File: templates/form-fields/file-uploaded.php [hunk 2]
```diff
@@ -19,23 +19,5 @@
 ?>
 
 <div class="wpum-uploaded-file">
-	<?php
-	if ( is_numeric( $data->value ) ) {
```
[the remark this answers is not in the document]
Comment (reygcalantaol) [reply templates/form-fields/file-uploaded.php:23]: @polevaultweb 
The purpose of this code was to preview the image with the remove button.
But this was already covered by filepond uploader.

File: assets/js/wp-user-manager.js (hunk 1 above)
Comment (reygcalantaol) [reply assets/js/wp-user-manager.js]: @polevaultweb 
This is done.

File: includes/functions.php [hunk 3]
```diff
@@ -564,9 +564,29 @@ function wpum_upload_file( $file, $args = array() ) {
...
+			if ( $wpum_uploading_file === 'user_avatar' || $wpum_uploading_file === 'user_cover' ) {
+
+				$default_name = basename( $upload['file'] );
+
+				//get size from filepond
+				$sizes = $args['sizes']['resize']['size'];
+				$sizes = apply_filters( 'wpum_sizes_'.$args['file_key'], $sizes, $upload );
+
+				if ( ! empty( $sizes ) ) {
+					$meta_data = image_make_intermediate_size( $upload['file'], $sizes['width'], $sizes['height'], false );
```
[the remark this answers is not in the document]
Comment (reygcalantaol) [reply includes/functions.php]: This is done.

File: includes/functions.php [hunk 4]
```diff
@@ -564,9 +564,29 @@ function wpum_upload_file( $file, $args = array() ) {
...
+
+			$file_url       = $upload['url'];
+			$file_path      = $upload['file'];
+
+			if ( $wpum_uploading_file === 'user_avatar' || $wpum_uploading_file === 'user_cover' ) {
+
+				$default_name = basename( $upload['file'] );
+
+				//get size from filepond
+				$sizes = $args['sizes']['resize']['size'];
```
[the remark this answers is not in the document]
Comment (reygcalantaol) [reply includes/functions.php]: This is fixed.

File: includes/wpum-forms/class-wpum-registration-forms-editor.php [hunk 5]
```diff
@@ -620,6 +624,48 @@ private function send_json_error() {
...
 
+	public function wpum_upload_profile_image(){
+		$allowed_mime_types = wpum_get_allowed_mime_types( 'current_' . $_POST['key'] );
+
+		$data = isset( $_POST[ $_POST['key'] ] ) ? json_decode( stripslashes( $_POST[ $_POST['key'] ] ), true, JSON_UNESCAPED_SLASHES) : [];
+
+		$uploaded = wpum_upload_file( $_FILES[ $_POST['key'] ], array(
+			'file_key'           => $_POST['key'],
+			'allowed_mime_types' => $allowed_mime_types,
+			'file_label'         => $field['label'],
```
[the remark this answers is not in the document]
Comment (reygcalantaol) [reply includes/wpum-forms/class-wpum-registration-forms-editor.php]: This is fixed.

File: includes/wpum-forms/class-wpum-registration-forms-editor.php [hunk 6]
```diff
@@ -37,6 +37,10 @@ public function init_hooks() {
 		add_action( 'wp_ajax_wpum_save_registration_form_settings', [ $this, 'save_form_settings' ] );
 		add_action( 'wp_ajax_wpum_get_registration_form_field', [ $this, 'get_form_field_data' ] );
 
+		add_action( 'wp_ajax_wpum_upload_profile_image', [ $this, 'wpum_upload_profile_image' ] );
```
[the remark this answers is not in the document]
Comment (reygcalantaol) [reply includes/wpum-forms/class-wpum-registration-forms-editor.php]: @polevaultweb 
This is done

File: templates/form-fields/file-field.php [hunk 7]
```diff
@@ -51,7 +51,7 @@
...
+<input type="file" class="<?php echo esc_attr( implode( ' ', $classes ) ); ?>" data-file_types="<?php echo esc_attr( implode( '|', $allowed_mime_types ) ); ?>" <?php if ( ! empty( $data->multiple ) ) echo 'multiple'; ?> name="<?php echo esc_attr( isset( $data->name ) ? $data->name : $data->key ); ?><?php if ( ! empty( $data->multiple ) ) echo '[]'; ?>" id="<?php echo esc_attr( $data->key ); ?>" placeholder="<?php echo empty( $data->placeholder ) ? '' : esc_attr( $data->placeholder ); ?>" data-file-key="<?php echo esc_attr( $data->key ); ?>" data-file-action="wpum_upload_profile_image"  data-file-nonce="<?php echo wp_create_nonce( 'verify_account_form' ); ?>" data-file-layout="<?php echo $data->key == 'user_avatar' ? 'compact circle' : ''; ?>" data-file-width="<?php echo $data->key == 'user_avatar' ? '250' : ''; ?>" data-file-height="<?php echo $data->key == 'user_avatar' ? '250' : ''; ?>" data-file-poster="<?php echo $data->value; ?>" data-file-posterkey="<?php echo $data->key == 'user_avatar' ? 'current_user_avatar_path' : 'user_cover_path'; ?>" />
```
Comment (polevaultweb) [inline templates/form-fields/file-field.php:54]: Getting this when visiting a registration form with a file field on it

 PHP Notice:  Undefined property: stdClass::$value in /Users/igp/sites/wpum/public/content/plugins/wp-user-manager/templates/form-fields/file-field.php on line 54

File: templates/form-fields/file-uploaded.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (polevaultweb) [reply templates/form-fields/file-uploaded.php:23]: Right, but there are some other file types that were previewed in different ways. Should keep that working. 

Can you test with a video and audio file to check

File: includes/actions.php [hunk 8]
```diff
@@ -371,3 +371,55 @@ function wpum_register_profile_privacy_fields() {
 	         ) );
 }
 add_action( 'carbon_fields_register_fields', 'wpum_register_profile_privacy_fields' );
+
+add_action( 'wp_ajax_wpum_upload_profile_image', 'wpum_upload_profile_image' );
```
Comment (polevaultweb) [inline includes/actions.php:375]: Getting a 400 error when uploading files on a registration form (user not logged in)

File: templates/form-fields/file-field.php (hunk 7 above)
Comment (polevaultweb) [inline templates/form-fields/file-field.php:54]: 250 for the image width and height needs to be filterable

File: includes/actions.php (hunk 8 above)
Comment (reygcalantaol) [reply includes/actions.php:375]: @polevaultweb 
I'll check.

File: assets/css/vendor/filepond/filepond-preview.css [hunk 9]
```diff
@@ -0,0 +1,163 @@
+/*!
```
Comment (polevaultweb) [inline assets/css/vendor/filepond/filepond-preview.css:1]: Is it possible to add custom styling to the image preview to make the green overlay blue?

image

Comment (reygcalantaol) [reply assets/css/vendor/filepond/filepond-preview.css:1]: @polevaultweb 
I've committed new changes for this.

File: templates/form-fields/file-field.php (hunk 7 above)
Comment (reygcalantaol) [reply templates/form-fields/file-field.php]: @polevaultweb 
I've committed new changes for this.

File: includes/actions.php (hunk 8 above)
Comment (reygcalantaol) [reply includes/actions.php]: @polevaultweb 
I've committed new changes for this.

File: templates/form-fields/file-field.php (hunk 7 above)
Comment (reygcalantaol) [reply templates/form-fields/file-field.php]: @polevaultweb 
I've committed new changes for this.

**Pull request closed without merging**